foundry/applications/formApplications/baseEntitySheets/rollTableConfig.d.ts

```diff
@@ -1,25 +1,151 @@
-// @TODO:
-
 /**
  * The RollTable configuration sheet
- *
- * @param table - The rollable table entity being configured
- * @param options - Additional application rendering options
  */
-declare class RollTableConfig extends BaseEntitySheet {
+declare class RollTableConfig extends BaseEntitySheet<RollTableConfig.Data, RollTable> {
+  static get defaultOptions(): RollTableConfig.Options;
+
   /**
-   * Return a dynamic application title for the configuration sheet
+   * @override
    */
   get title(): string;
 
   /**
-   * Activate event listeners for the RollTable Configuration Sheet
+   * @param options - (unused)
+   * @override
+   */
+  getData(options?: Application.RenderOptions): RollTableConfig.Data;
+
+  /**
+   * @override
    */
   activateListeners(html: JQuery): void;
 
   /**
-   * @param options - (unused)
+   * Handle creating a TableResult in the RollTable entity
+   * @param event      - The originating mouse event
+   * @param resultData - An optional object of result data to use
+   */
+  protected _onCreateResult(event: MouseEvent, resultData?: object): Promise<any>; // TODO: type when TableResult is typed
+
+  /**
+   * Submit the entire form when a table result type is changed, in case there are other active changes
+   */
+  protected _onChangeResultType(event: Event): ReturnType<RollTableConfig['_onSubmit']>;
+
+  /**
+   * Handle deleting a TableResult from the RollTable entity
+   */
+  protected _onDeleteResult(event: Event): Promise<any>; // TODO: type when TableResult is typed
+
+  /**
    * @override
    */
-  getData(options?: any): BaseEntitySheet.Data<Entity>;
+  protected _onDrop(event: DragEvent): ReturnType<RollTableConfig['_onCreateResult']>;
+
+  /**
+   * Handle changing the actor profile image by opening a FilePicker
+   */
+  protected _onEditImage(event: Event): ReturnType<FilePicker['browse']>;
+
+  /**
+   * Handle a button click to re-normalize dice result ranges across all RollTable results
+   */
+  protected _onNormalizeResults(event: Event): Promise<any>; // TODO: type when TableResult is typed
+
+  /**
+   * Handle toggling the drawn status of the result in the table
+   */
+  protected _onLockResult(event: Event): Promise<any>; // TODO: type when TableResult is typed
+
+  /**
+   * Reset the Table to it's original composition with all options unlocked
+   */
+  protected _onResetTable(event: Event): Promise<any>; // TODO: type when TableResult is typed
```

Comment:
event is always a `JQuery.ClickEvent` (might not be needed), should return a Promise<RollTable>.